libs/confman/src/file_storage.cpp

```diff
@@ -0,0 +1,141 @@
+#include <cstdint>
+#include <cstdlib>
+#include <iterator>
+#include <map>
+#include <set>
+#include <system_error>
+
+#include <stduuid/uuid.h>
+
+#include <tfc/utils/pragmas.hpp>
+
+// clang-format off
+// dynamic and clang build would fail if this warning is not deactivated
+PRAGMA_CLANG_WARNING_PUSH_OFF(-Wdate-time)
+#include <pcg_random.hpp>
+#include <pcg_extras.hpp>
+PRAGMA_CLANG_WARNING_POP
+// clang-format on
+
+#include <tfc/confman/file_storage.hpp>
+
+namespace tfc::confman {
+
+/// \brief Retrieves an integer value of an environment variable, if a non-negative value is present on the variable
+/// it is returned. Otherwise a default value is returned.
+/// \param env_variable environment variable to fetch
+/// \param default_value default integer value to be returned if the environment variable is not set or holds an invalid
+/// value. \return value of the environment variable or the default value. \example
+/// get_environment_variable("MAX_CONNECTIONS", 5); If the environment variable "MAX_CONNECTIONS" is set to "10", the
+/// function returns 10. Otherwise, the function returns 5.
+static auto get_environment_variable(std::string const& env_variable, int default_value) -> int {
+  if (const char* value = std::getenv(env_variable.c_str()); value != nullptr) {
+    if (int const value_i = std::stoi(value); value_i >= 0) {
+      return value_i;
+    }
+  }
+  return default_value;
+}
```

Comment:
You should rename this or split into two. The "hidden" conversion to an int based on the name is not great.

Why is value_i of 0 invalid?
maybe you would like to store 0 DAYS but always 10 copies.

std::stoi can throw. Let's catch that and log rather then crashing.
see https://en.cppreference.com/w/cpp/string/basic_string/stol

Reply:
`value_i` can be 0, `value_i >= 0` means equal to or greater than 0.